janus_server/src/bin/aggregation_job_creator.rs

```diff
@@ -1,26 +1,35 @@
+use anyhow::Context;
 use janus_core::time::RealClock;
 use janus_server::aggregator::aggregation_job_creator::AggregationJobCreator;
-use janus_server::binary_utils::{janus_main, BinaryOptions, CommonBinaryOptions};
+use janus_server::binary_utils::{
+    janus_main, setup_signal_handler, BinaryOptions, CommonBinaryOptions,
+};
 use janus_server::config::{BinaryConfig, CommonConfig};
 use serde::{Deserialize, Serialize};
 use std::sync::Arc;
 use std::time::Duration;
 use structopt::StructOpt;
+use tokio::select;
 
 #[tokio::main]
 async fn main() -> anyhow::Result<()> {
     janus_main::<_, Options, Config, _, _>(RealClock::default(), |ctx| async move {
+        let shutdown_signal =
+            setup_signal_handler().context("failed to register SIGTERM signal handler")?;
+
```

Comment:
I'd prefer not to, as the `aggregator` currently does a [graceful shutdown](https://hyper.rs/guides/server/graceful-shutdown/) rather than just cancelling the main future, dropping everything, and exiting. Unifying on cancellation would mean we'd have to drop graceful shutdown, and we could wind up giving more clients connection resets or 503s than otherwise.

Reply:
I see, I hadn't noticed there was a difference in behavior between the two.